index.js

```diff
@@ -21,61 +26,15 @@ if (typeof applyMiddleware === 'function') {
   applyMiddleware(app);
 }
 
-if (process.env.NODE_ENV === 'production') {
-  const path = require('path');
-  const expressStaticGzip = require('express-static-gzip');
-  const serverRenderer = require('./dist/server.js').default;
-  const clientStats = require('./dist/client-stats.json');
-  const distPath = path.join(__dirname, 'dist');
-  app.use(
-    // This path should be in sync with the `publicPath` from webpack config.
-    `${env.BASE_URL}/assets`,
-    expressStaticGzip(distPath, {
-      maxAge: '1d',
-    }),
-  );
-  app.use(serverRenderer({ clientStats, currentDirectory: __dirname }));
-} else {
-  process.env.NODE_ENV = 'development';
-  console.log('Server is running in development mode');
-  const webpack = require('webpack');
-  const webpackDevMiddleware = require('webpack-dev-middleware');
-  const webpackHotMiddleWare = require('webpack-hot-middleware');
-  const webpackHotServerMiddleware = require('webpack-hot-server-middleware');
-
-  require('ts-node').register({
-    compilerOptions: { module: 'commonjs' },
-    cacheDirectory: '/tmp',
-  });
-  const { default: configs, clientConfig } = require('./webpack.config');
-
-  const compiler = webpack(
-    configs.map((c) =>
-      Object.assign({}, c, {
-        mode: 'development',
-      }),
-    ),
-  );
-  const clientCompiler = compiler.compilers.find(({ name }) => name === 'client');
-
-  const devMiddleware = webpackDevMiddleware(compiler, {
-    serverSideRender: true,
-    publicPath: clientConfig.output.publicPath,
-  });
-
-  app.use(devMiddleware);
-  app.use(webpackHotMiddleWare(clientCompiler));
-  app.use(
-    webpackHotServerMiddleware(compiler, {
-      serverRendererOptions: {
-        // Send client compiler FS for reading index.html for emitted assets
-        fileSystem: clientCompiler.outputFileSystem,
-        // Helps with finding the output folder in memory-fs
-        currentDirectory: __dirname,
-      },
-    }),
-  );
-}
+const distPath = path.join(__dirname, 'dist');
+app.use(
+  // This path should be in sync with the `publicPath` from webpack config.
+  `${env.BASE_URL}/assets`,
```

Comment:
nit: as this path should be in sync with path in webpack config, maybe it makes sense to create a separate variable for it (maybe in env file, as both this files already require it and Base_URL comes from there)